src/device/mod.rs

```diff
@@ -1997,6 +1998,32 @@ impl TaskRuntime for Runtime {
             },
 
             Some(mesh_event) = self.event_listeners.wg_event_subscriber.recv() => {
+                let public_key = mesh_event.peer.public_key;
+
+                if let Some(mesh_entities) = self.entities.meshnet.as_ref() {
+                    if let Some(proxy_endpoints) = mesh_entities.proxy.get_endpoint_map().await.ok().and_then(|proxy_map| proxy_map.get(&public_key).cloned()) {
+                        let is_proxying = mesh_event.peer.endpoint.map(|ep| proxy_endpoints.contains(&ep)).unwrap_or(false);
+                        let was_proxying = mesh_event.old_peer.and_then(|peer| peer.endpoint.map(|ep| proxy_endpoints.contains(&ep))).unwrap_or(false);
```

Comment:
Nitpick: `map()` and then `unwrap_or()` can be shortened with single call `map_or()`

Reply:
In the first one it makes sense, in the second, as I have two layers of options, the current one looks better to me.